### DISCUSSION TITLE
slow indexing on a big dbase ; possible damaging of the files searched

### DISCUSSION THREAD

**Discussion by cyber-soho** [cboxdoerfer/fsearch#478] [Q&A]
Hello, Everyone.
I hope all is well with you.
I am using fsearch with Mint 21.1, and set it up to update the data- base everytime its starts, and since there are couple of TB in my Debian- based nas, everytime it s taking a lot to update.
I saw in #474 that the Linux FS has some limitations to speed up the indexing, but if there is way around anyways (of course, I can always choose to manually update the data- base at a later time) ?
Also, I am trying to access, and update the database as a nas- user, with less privileges than the nas- administrator, but I am interested if the indexing itself might cause some unexpected damage over some or all files into the nas ?
Thanks, and Regards
CS

**Comment by cboxdoerfer** [comment]
Hi,

> I am using fsearch with Mint 21.1, and set it up to update the data- base everytime its starts, and since there are couple of TB in my Debian- based nas, everytime it s taking a lot to update.

Unfortunately this can't be avoided on Linux. The only way to know which files have been changed since the last time FSearch was running is by scanning all the folders again.

The upcoming filesystem monitoring won't change that, because before FSearch can even start monitoring a folder, it first needs to scan it.

> Also, I am trying to access, and update the database as a nas- user, with less privileges than the nas- administrator, but I am interested if the indexing itself might cause some unexpected damage over some or all files into the nas ?

No, the only file which gets modified during indexing is the database file itself; everything else is a read-only operation.

**Reply by cyber-soho** [reply]
Thank you for your prompt reply, Christian.
Where indeed will be data- base file located in the Mint FS ?
Best,
CS

**Reply by cboxdoerfer** [reply]
No problem. The database file can be found at `$XDG_DATA_HOME/fsearch/fsearch.db`. Typically that's `~/.local/share/fsearch/fsearch.db`.

**Reply by cyber-soho** [reply]
Awesome, thank you so much, Christian !
Best,
CS
